### PR TITLE
Fleshed out a code of conduct

### PR DESCRIPTION
**Pull request by mjbrender**
Heavily borrowed from known good definitions. Addresses #48.

### CODE REVIEW

**Comment by mjbrender** [comment]
Questions for reviewers: 

**High level:**
* Is breaking it out into "Specific Conduct" and "Additional Community Beliefs" a reasonable idea?

**Lower level:** 
* Are there specific rules of conduct that are missing? 
* Are the specific highlighted ideas, such as pragmatism, right for the culture we're choosing?

**Comment by danieldreier** [comment]
The content and organization seems reasonable to me. Thanks for working on this. One suggestion - I like that [geek feminism's suggested policy](http://geekfeminism.wikia.com/wiki/Community_anti-harassment/Policy) includes "Participants are responsible for knowing and abiding by these rules."

**Comment by mjbrender** [comment]
@danieldreier good call. I added that wording and a few more clarifications. I'm feeling good with where we stand here and will slot this for the next release.

File: code-of-conduct.md [hunk 1]
```diff
@@ -1,5 +1,89 @@
...
+
+
+## Specific Conduct
+
+Participants are responsible for knowing and abiding by the following:
+
+* We **err on assuming contributors and users mean well**. This assumption enables us to discuss anything in the open in a mutually respectful way.
+* Whether you're a regular contributor or a newcomer, all members care about making this community a safe place for you and we've got your back.
+* Respect that people have differences of opinion and that every design or implementation choice carries a trade-off and numerous costs. There is seldom a right answer.
+* Please keep unstructured critique to a minimum. If you have solid ideas you want to experiment with, make a fork and see how it works.
```
Comment (laurenrother) [inline code-of-conduct.md]: I'm not sure what "unstructured critique" means exactly. I'm not sure whether it just needs a clearer definition or link to an outside resource?

File: code-of-conduct.md [hunk 2]
```diff
@@ -1,5 +1,89 @@
 #Code of Conduct
 
-TBD. For now, err on the side of kindness.
+As members of this Community, we recognize that our behavior is our choice and will result in our culture.
```
Comment (laurenrother) [inline code-of-conduct.md]: "As members of the Basho Community, we recognize that our behavior is our choice and that this choice will determine our culture."

File: code-of-conduct.md (hunk 1 above)
Comment (mjbrender) [inline code-of-conduct.md]: Good catch @laurenrother. I don't find that sentence valuable on second review and I'll remove it.

**Comment by mjbrender** [comment]
I'm putting this out into the wild. It's a good starting place and can be improved by our community going forward.

**Pull request merged**